v3/integration/genTestCerts/genTestCerts.go

```diff
@@ -0,0 +1,314 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Generates a CA, an intermediate, and a leaf certificate and prints their
+// OpenSSL textual output to stdout.
+func main() {
+	ca, err := NewTrustAnchor()
+	if err != nil {
+		panic(err)
+	}
+	fmted, err := OpenSSLFormatCertificate(ca)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("-------------Trust Anchor-------------")
+	fmt.Println(fmted)
+	intermediate, err := NewIntermediate(ca)
+	if err != nil {
+		panic(err)
+	}
+	fmted, err = OpenSSLFormatCertificate(intermediate)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("-------------Intermediate-------------")
+	fmt.Println(fmted)
+	leaf, err := NewLeaf(ca, []*x509.Certificate{intermediate})
+	if err != nil {
+		panic(err)
+	}
+	fmted, err = OpenSSLFormatCertificate(leaf)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("-------------Leaf-------------")
+	fmt.Println(fmted)
+}
+
+// This is NOT a healthy example of a CA certificate, this is nothing
+// more than a self signed certificate with IsCA set to true. Not even any
+// basic constraints are defined. Please do not think that this will be
+// acceptable to any system, let alone lint particularly well.
+func NewTrustAnchor() (*x509.Certificate, error) {
```

Comment:
Across the board I'd encourage not exporting any funcs since they're part of the internal API for the script.

Reply:
Agreed, although this is a `package main` and I don't think others could import these names even they wanted to (I don't think you can import `example.com/project/main`).

But sure. Perhaps I just don't like camel case and that's why I do this for self contained scripts =p

Reply:
> (I don't think you can import example.com/project/main).

Fair point!